apps/projects/app/components/Panel/ReviewWork/ReviewWork.js

```diff
@@ -58,6 +87,10 @@ const ReviewWork = ({ issue, submissionIndex, readOnly }) => {
   const updateRating = (index) => setRating(index)
   const updateFeedback = e => setFeedback(e.target.value)
   const changeSubmission = (index) => setIndex(index)
+  const changeRequestType = (index) => {
+    setIndex(0)
+    setIndexType(index)
+  }
```

Comment:
It looks like all the logic to define `typesToShow`, `indexTypeSelected`, `workSubmissions`, `newSubmissionIndex`, and even `changeRequestType` is _super_ similar to the logic in the `ReviewApplication` panel. If we can pull it into a hook, I'd be surprised if we wouldn't be able to use the same hook in both spots. Maybe it will end up looking something like this:

```js
const {
  filters,
  selectedFilter,
  filteredItems,
  selectedItem,
  selectFilter,
  selectItem,
} = useReviewFilters({
  allItems: issue.requestsData, // or issue.workSubmissions
  initialSelection: issue.requestsData[submissionIndex], // or issue.workSubmissions[submissionIndex]
})
```

(The `initialSelection` thing will only work if my second suggestion for setting `selectedItem` works out. If not, you may have to pass two separate variables. But there should still be a way to do this!)

Reply:
Done!